coordinator/persistence/aggregation/src/integrationTest/kotlin/net/consensys/zkevm/persistence/dao/aggregation/AggregationsPostgresDaoTest.kt

```diff
@@ -293,6 +293,90 @@ class AggregationsPostgresDaoTest : CleanDbTestSuiteParallel() {
     }
   }
 
+  @Test
+  fun findConsecutiveProvenBlobsReturnsFullListOfProvenBlobsWithProvenBatches() {
+    val batch1 = createBatch(0L, 10L)
+    val batch2 = createBatch(11L, 20L)
+    val batch3 = createBatch(21L, 30L)
+
+    val blob1 = createBlobRecordFromBatches(listOf(batch1))
+    val blob2 = createBlobRecordFromBatches(listOf(batch2, batch3))
+
+    val batches = listOf(batch1, batch2, batch3)
+    val blobs = listOf(blob1, blob2)
+
+    batches.forEach { insertBatch(it).get() }
+    blobs.forEach { insertBlob(it).get() }
+
+    aggregationsPostgresDaoImpl.findConsecutiveProvenBlobs(0).get().also { blobCounters ->
+      assertThat(blobCounters).hasSameElementsAs(
+        listOf(
+          getBlobAndBatchCounters(listOf(batch1), blob1),
+          getBlobAndBatchCounters(listOf(batch2, batch3), blob2)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun findConsecutiveProvenBlobsWhenDbIsEmpty() {
+    aggregationsPostgresDaoImpl.findConsecutiveProvenBlobs(0).get().also { blobCounters ->
+      assertThat(blobCounters).hasSameElementsAs(emptyList())
+    }
+  }
+
+  @Test
+  fun findConsecutiveProvenBlobsUnprovenBlobAndBatchAtDifferentIntervals() {
```

Comment:
I thought we were not supposed to have unproven records in the DB. Are we trying to represent any real scenario here?

However, if the DAO/db was designed for such scenarios, then it shall test it

Reply:
Yes, the Dao and db were designed at the beginning to support unproven records